### DISCUSSION TITLE
Nomenclature for default Clip names could be improved as "Audio Clip"

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#1535] [Development]
I have been testing Vitaly's latest alpha build for 3.1.0  with Clip names added - Audacity_Windows_64bit_1167467960_f53c93b 

The default name for a Clip is "**Audio Track**" - this is misleading and could be more informative:
a) the clip is not a track - especially clear when you have multiple clips in a track
b) it misses the UX/GUI opportunity to identify the clip as a "Clip"

My recommendation is that the default clip name should be "**Audio Clip**"

This then clearly introduces the user to the concept of clips and naming of clips.

**Comment by vsverchinsky** [comment]
The 1st case acutally a bug, this is fixed in the latest build. There was an agreement that clips for imported tracks should have same name as a track itself

**Reply by petersampsonaudacity** [reply]
> The 1st case acutally a bug, this is fixed in the latest build. There was an agreement that clips for imported tracks should have same name as a track itself

Ah that would seem to make sense - I think I was actually expecting that.

**Comment by petersampsonaudacity** [comment]
Another interesting "feature" of clip names is that they can easily disappear off the screen when you zoom or scroll.  Thus leaving what looks like an un-named clip. 

This is presumably because the clip label is "stuck" to the extreme left of the clip.

Here you can see a "partial eclipse" of the clip name:
![image](https://user-images.githubusercontent.com/50205984/130982820-e5531cd0-48b6-43f2-a6d8-83c6c37ecad0.png)


Perhaps:
**IF** a clip is at all visible on screen
**THEN** its label should always be displayed

**Reply by DavidBailes** [reply]
> **IF** a clip is at all visible on screen
> **THEN** its label should always be displayed

if the clip is partially visible in this situation, then Reaper displays the name, and prefixes it with << to indicate that the clip is partially hidden.

**Comment by DavidBailes** [comment]
> 5\. This looks like a most odd and confusing numbering system to me (as a user)

and difficult to imagine this being of much use to anyone.

**Comment by SteveDaulton** [comment]
I don't "get" what these clip names are for.

When I create (record or generate) a new clip in an audio track, it gets the name of the audio track + one or more numbers. This seems like it could potentially be useful.

If I copy the clip into a new track, the clip name remains the same - it doesn't get the new track name.
I can see how it might be useful for the clip names to be "permanent", allowing us  to track where the clips came from - except that the names are not really permanent. If I mix and render a track, the clip names are lost as the rendered track now has one clip with the track name.

And then it gets more bizarre:
If I have a "Guitar" track and a "Vocal" track:
![Tracks000](https://user-images.githubusercontent.com/11681097/132662077-b3c19ff2-e84b-4a3c-9bc3-440ede8fa2dd.png)

Copy the highlighted section from the Guitar track and paste it into the Vocal track:
![Tracks001](https://user-images.githubusercontent.com/11681097/132662192-271eb78f-7ffe-4123-b9ba-868d54f3f2ef.png)

As expected, the "Guitar" clip is merged into the "Vocal" clip.
But what if I paste at the start of the Vocal clip?
![Tracks002](https://user-images.githubusercontent.com/11681097/132662395-a535f745-b37c-415a-8060-2e05c1a8bfbb.png)

 Why is the name of the clip different? Is it intuitive? Is it useful?

**Reply by vsverchinsky** [reply]
Didn't expect that, seems like unpredicted consequence (I mean the last case). Font size is quite large, what OS is it?

**Reply by SteveDaulton** [reply]
Xubuntu 20.04.
My font settings:
![fonts](https://user-images.githubusercontent.com/11681097/132681464-8f0c8ca3-3c4f-4bb1-804d-be1902152e07.png)

**Reply by vsverchinsky** [reply]
Thanks!

**Comment by SteveDaulton** [comment]
Imo, automatic clip naming needs to be thought through again.

![Tracks003](https://user-images.githubusercontent.com/11681097/132666136-2e6a1b2e-1944-42ed-b9a4-b093949f2065.png)

**Comment by petersampsonaudacity** [comment]
I still think  the default clip name would be **much** better and **much** clearer in intent/purpose it it were

**Audio Clip** rather than Audio Track

And as I said earlier it would make clips and their purpose more "discoverable"

Just look at this example:
![image](https://user-images.githubusercontent.com/50205984/136829174-f4c44b31-d544-4bd0-9231-e27183af4593.png)
The four things in that audio track are clips they are not audio tracks.

@Tantacrul this is a UX/GUI design issue surely ...

**Reply by SteveDaulton** [reply]
> I still think the default clip name would be **much** better and **much** clearer in intent/purpose it it were
> **Audio Clip** rather than Audio Track

but isn't the idea that the clip name says which track the clip came from?
If I import a track called "808 drums", are you suggesting that the clip should be called "Audio Clip" rather than "808 drums"?

**Reply by petersampsonaudacity** [reply]
> but isn't the idea that the clip name says which track the clip came from?
> If I import a track called "808 drums", are you suggesting that the clip should be called "Audio Clip" rather than "808 drums"?

It's a fair cop guv'nor   ;-)

**Comment by petersampsonaudacity** [comment]
See also  Default clip titles should not be named "Audio Track" #2499

**Reply by LWinterberg** [reply]
(I've hidden some of the earlier comments here as they got addressed by #1698)

**Comment by LWinterberg** [comment]
The easiest way probably would be to change the default from "Audio Track" to "Audio", optionally adding "clip" to each clip afterwards. That way, you'd get something like `Audio #1` or `Audio Clip #1`, both of which probably are sufficiently clear. When importing a file, it'd be `Filename` or `Filename Clip` (though I guess there could be a rule that just omits the `clip` when importing files). 

That said, I wonder if there's a good way (or even a use case) for a more sophisticated solution.

**Reply by SteveDaulton** [reply]
Why not just have a default name for clips (such as "Audio Clip").
```
If (track-name == default-track-name)
   clip-name = default-clip-name;
else
   clip-name = track-name;
```
